turbopack/crates/turbopack-core/src/resolve/pattern.rs

```diff
@@ -1179,7 +1178,9 @@ impl Pattern {
 impl Pattern {
     #[turbo_tasks::function]
     fn new_internal(pattern: Value<Pattern>) -> Vc<Self> {
-        Self::cell(pattern.into_value())
+        let mut pattern = pattern.into_value();
+        pattern.normalize();
```

Comment:
The normalization need to be applied in `Pattern::new`. Otherwise a pattern and the normalized version of the pattern have a different Vc, breaking the singleton pattern.